homework22/src/ba/bitcamp/homework/networking/Server.java

```diff
@@ -0,0 +1,140 @@
+package ba.bitcamp.homework.networking;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URL;
+
+/**
+ * This class present local server which get inputed link and name of link from
+ * connected server, then check is link valid, and write valid link in the file
+ * 
+ * @author Niddal
+ *
+ */
+public class Server {
+
+	static final int PORT = 3333;
+
+	public static void main(String[] args) {
+
+		String line = "";
+
+		try {
+			System.out.println("Server is made, waiting for client...");
+			@SuppressWarnings("resource")
+			ServerSocket server = new ServerSocket(PORT);
+			Socket client = server.accept();
+			System.out.println("Client is connected!!!");
+			while (true) {
+				BufferedReader reader = new BufferedReader(
+						new InputStreamReader(client.getInputStream()));
+
+				line = reader.readLine();
+
+				System.out.println(line);
+
+				System.out.println(getLink(line));
+				System.out.println(getLinkName(line));
+
+				File file = new File("info.txt");
+				if (!file.exists()) {
+					file.createNewFile();
+				}
+
+				String link = getLink(line);
+				// System.out.println(getResponseCode(link));
+				System.out.println(isValid(link));
+				@SuppressWarnings("resource")
+				BufferedWriter writer = new BufferedWriter(new FileWriter(file,
+						true));
+				if (isValid(link)) {
+
+					writer.write(line);
+					writer.newLine();
+					writer.flush();
+					line = " ";
+
+				} else {
+					line = " ";
+				}
```

Comment:
Posto se u oba slucaja line = " " izvrsava, moglo se jednom napisati bez obzira na uslov. Takodjer, ne bih preporucila nikad instanciranje String objekta na prazan string, jer ne vidim nijedan slucaj gdje bi ovo bilo korisno. Ako nesto ne postoji, poruka treba to reci ili string treba biti null pa se taj slcaj opet negdje handlati.